src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -658,6 +658,36 @@ public ProjectState UpdateAdditionalDocument(TextDocumentState newDocument, bool
                 latestDocumentTopLevelChangeVersion: dependentSemanticVersion);
         }
 
+        public ProjectState UpdateDocumentsOrder(IEnumerable<DocumentId> documentIds)
+        {
+            var orderedDocumentIds = ImmutableList.CreateBuilder<DocumentId>();
+
+            foreach(var documentId in documentIds)
+            {
+                if (this.ContainsDocument(documentId))
```

Comment:
when would this be false?

Reply:
If you only passed a partial list of source files to order, you need to retain remaining documents that were not part of the order.